core/services/ocr2/plugins/promwrapper/plugin.go

```diff
@@ -223,13 +232,17 @@ func New(
 	}
 
 	return &promPlugin{
-		wrapped:           plugin,
-		name:              name,
-		chainType:         chainType,
-		chainID:           chainID,
-		oracleID:          fmt.Sprintf("%d", config.OracleID),
-		configDigest:      common.Bytes2Hex(config.ConfigDigest[:]),
-		prometheusBackend: prometheusBackend,
+		wrapped:                       plugin,
+		name:                          name,
+		chainType:                     chainType,
+		chainID:                       chainID,
+		oracleID:                      fmt.Sprintf("%d", config.OracleID),
+		configDigest:                  common.Bytes2Hex(config.ConfigDigest[:]),
+		prometheusBackend:             prometheusBackend,
+		queryEndTimes:                 cache.New(DefaultExpiration, DefaultCleanupInterval),
+		observationEndTimes:           cache.New(DefaultExpiration, DefaultCleanupInterval),
+		reportEndTimes:                cache.New(DefaultExpiration, DefaultCleanupInterval),
+		acceptFinalizedReportEndTimes: cache.New(DefaultExpiration, DefaultCleanupInterval),
```

Comment:
out of curiosity, is there a reason we can't use the built-in [histogram](https://prometheus.io/docs/practices/histograms/) type to collect this sort of data with a fixed memory size?

Reply:
The histogram itself can't measure time between two points. If I am getting it correctly, it's just a collection of counters that record how many observations fall into predefined duration buckets. We need the timestamp cache to actually calculate the elapsed time between one OCR step and the next one.

Reply:
Ok, I thought it was measuring the steps themselves not the time between steps. That makes it more complicated.